services/management/adapter/file_provider.go

```diff
@@ -169,7 +169,14 @@ func (mp *FileProvider) parseData(contents []byte) (*management.VirtualChainMana
 	vcString := fmt.Sprintf("%d", mp.config.VirtualChainId())
 	vcData, ok := data.VirtualChains[vcString]
 	if !ok {
-		return nil, errors.Errorf("could not find current vc in data")
+		return nil, errors.Errorf("could not find current vc in data (%d)", mp.config.VirtualChainId())
+	}
+
+	if data.CurrentRefTime != 0 {
+		if data.CurrentRefTime < data.PageStartRefTime || data.CurrentRefTime > data.PageEndRefTime {
```

Comment:
`data.CurrentRefTime > data.PageEndRefTime` if we have historic polling dont we want current to be greater than page end?...